src/util/event_emitter.js

```diff
@@ -0,0 +1,140 @@
+export class EventEmitter {
+  constructor() {
+    this._handlers = [];
+  }
+
+  /**
+   * Adds the event name and handler function to the end of the handlers array.
+   * No checks are made to see if the handler has already been added.
+   * Multiple calls passing the same combination of event name and handler will result in the handler being added,
+   * and called, multiple times.
+   * @param {String} event
+   * @param {Function} handler
+   * @returns {EventEmitter}
+   */
+  on(event, handler) {
+    this._handlers.push({
+      event,
+      handler
+    });
+
+    return this;
+  }
+
+  _onceWrapper(...args) {
+    if (!this.fired) {
+      this.target.off(this.event, this.wrapFn);
+      this.fired = true;
+      Function.prototype.apply.call(this.handler, this.target, [...args]);
+    }
+  }
+
+  _onceWrap(event, handler) {
+    const state = {
+      fired: false,
+      wrapFn: undefined,
+      target: this,
+      event,
+      handler
+    };
+    const wrapped = this._onceWrapper.bind(state);
+    wrapped.handler = handler;
+    state.wrapFn = wrapped;
```

Comment:
Why not set this when initialising the state ?

Reply:
I used the example from https://github.com/Gozala/events/blob/master/events.js . I'm also not sure why it is reassigned, I think it is because of the scope change?

Reply:
I removed it and simplified the once wrap functions